### DISCUSSION TITLE
Inclusion in official repositories

### DISCUSSION THREAD

**Discussion by aminvakil** [CyberShadow/aconfmgr#147] [Ideas]
Hi, thanks for this project, I really liked this repository and brought this up on #archlinux-aur channel on IRC Libera Chat network to see if a TU liked it too and wants to package it and put it into community.

A TU stated there is not releases and therefore maintaining it would be hard for them.

There is also no signed commit which one can use at least.

What do you think about creating signed tags and releases?

**Comment by CyberShadow** [comment]
That wouldn't be a problem.

Though, if the goal is to have it packaged in the official community repository, I suspect that it will not go through because aconfmgr can also act as an AUR helper (aconfmgr assumes that foreign packages need to be installed from the AUR, and to accomplish this, it either invokes a user-configured AUR helper or does it by itself). AUR helpers have historically never been packaged in official repositories.

If the forces that be decide to make an exception for aconfmgr (perhaps on the basis that the AUR functionality is not its primary purpose), I'll be happy to cut a release.

**Reply by aminvakil** [reply]
> If the forces that be decide to make an exception for aconfmgr (perhaps on the basis that the AUR functionality is not its primary purpose), I'll be happy to cut a release.

I don't think an expection can be made for this package, I'll bring this up though.

**Reply by CyberShadow** [reply]
Conceivably, with some work, the AUR functionality could be extracted into a "plugin", which could be installable separately from the AUR.

I think integration with AUR helpers would not run afoul of those rules, as one would still need to go to the AUR to install an AUR helper before they could use it, so we might be able to keep that in the main package.

Was there any conclusion in the discussion with TUs?

**Reply by aminvakil** [reply]
> Was there any conclusion in the discussion with TUs?

Let me paste the whole conversation:
```
13:24 <aminvakil> hi! i've just saw https://github.com/CyberShadow/aconfmgr and there is also https://aur.archlinux.org/packages/aconfmgr-git, i'm bringing this up here to see if a TU wants to move this to community
13:41 <grawlinson> >no releases published
13:46 <aminvakil> grawlinson: yes, and there is also no signed commit too
13:52 <aminvakil> grawlinson: I brought this up here (https://github.com/CyberShadow/aconfmgr/discussions/147)
13:58 <grawlinson> goodie
13:58 <grawlinson> is it just a set of bash scripts?
14:00 <grawlinson> the optional dependencies on aur helpers is a bit problematic
15:37 <aminvakil> grawlinson: aside from optdepends on aur helpers, it can act as an aur helper itself too, if there isn't an aur helper installed it will install the aur package itself
15:38 <aminvakil> i don't think this fits into official repositories then
```
And it was ended.

If the main package which would be installed from community repository does not install packages from AUR, it does not violate any rules IMO and anyone who wants that feature can use the plugin from AUR.

cc @grawlinson

**Reply by CyberShadow** [reply]
Thanks!

> 13:46 <aminvakil> grawlinson: yes, and there is also no signed commit too

For the record, the commits I make from my main PC are signed (and of course merge commits are signed by GitHub):

https://github.com/CyberShadow/aconfmgr/commits/master

Automatic commits which bump versions are not signed (and I'm not sure if it even makes sense to sign them, as the intent of signing / verifying them will not be the same as for commits made by a human maintainer).

**Reply by izmyname** [reply]
I'm a little bit late to the party, but - perhaps, splitting the package and make something like aconfmgr-aur for handling 'foreign' packages would be a good solution. Having aconfmgr in the official repos would simplify a new installation.

**offtop:** Also, I think it's a great and revolutionary piece of software. Wish it were more popular, though, as _'Arch Linux declarative config'_ query in various search engines doesn't find `aconfmgr`. I wouldn't find it if I didn't dive deep into comments of some 7 years old reddit discussion.
